frontend/public/components/droppable-edit-yaml.tsx

```diff
@@ -83,7 +83,7 @@ export const DroppableEditYAML = withDragDropContext<DroppableEditYAMLProps>(
       if (file.size <= maxFileUploadSize) {
         const reader = new FileReader();
         reader.onload = () => {
-          const buffer = Buffer.from(reader.result);
+          const buffer = Buffer.from(reader.result as ArrayBuffer);
```

Comment:
Hmm, do you know why this is needed?

This code is running in a browser and not node, so I wouldn't expect the node types to have an effect here.

Reply:
I think that `Buffer` is part of the node.js standard library and we're just including it in the build: 

https://github.com/nodejs/node/blob/e8cf3a6a972fb074411c6ce504e2d43a74ac2dff/lib/buffer.js#L297

Reply:
I was trying to see if we could remove references to node.js `Buffer` here, but it looks like that's what the istextorbinary library expects. I'm not crazy that we're fibbing about the type since `reader.result` is presumably not an `ArrayBuffer` since we need `as`. Some related to discussion in https://github.com/bevry/istextorbinary/issues/292

I don't have a good suggestion to fix, so I guess we can leave it. Worth a comment explaining I think.

@TheRealJon FYI since you worked on this previously

Reply:
I think it should be fine since we only use `reader.readAsArrayBuffer` for this FileReader instance

Reply:
It looks like `reader.result` _is_ actually an `ArrayBuffer`. Is the `as` needed because it could be a string as well depending on the read operation used?

https://developer.mozilla.org/en-US/docs/Web/API/FileReader/result

Reply:
Yeah, that's probably what it is, and TypeScript probably just doesn't know that

Reply:
Thanks, I think this is fine. Probably worth a comment explaining.

Reply:
Comment added!